cpp/include/raft/spectral/lanczos.hpp

```diff
@@ -0,0 +1,1228 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
```

Comment:
It would also be nice to see this placed in a more shared location / namspace. `raft/linalg/lanczos.hpp`? (Really just thinking of Scipy's [eigsh](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.eigsh.html))

Reply:
Speaking of which- it would be really useful to have a `lobpcg` primitive available at some point. cc @afender

Reply:
Please specify the desired location to create and move to. `raft/linalg/lanczos.hpp` confirmed?

Reply:
Sure, I think `raft/linalg/lanczos.hpp` is fine. We could always move it down the road if we need to.

Reply:
Done.